tasks/common/protocol-versions.md

```diff
@@ -0,0 +1,62 @@
+# `ProtocolVersions` State Changes
+
+The `ProtocolVersions` contract signals the _recommended_ and _required_ protocol versions to nodes.
+Halting on incompatible protocol vesions is currently opt-in.
+
+It has two functions `setRecommended` and `setRequired` to set both versions individually.
+We usually aim to set the _recommended_ version *2 weeks* before a protocol upgrade and the _required_
+version *2 days* before.
+
+## Deployments
+
+* Mainnet: [`0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A`](https://etherscan.io/address/0x9BA6e03D8B90dE867373Db8cF1A58d2F7F006b3A)
+* Sepolia: [`0x79ADD5713B383DAa0a138d3C4780C7A1804a8090`](https://sepolia.etherscan.io/address/0x79ADD5713B383DAa0a138d3C4780C7A1804a8090)
+
+## Storage Slots
+
+It has two storage slots:
+* `0x4aaefe95bd84fd3f32700cf3b7566bc944b73138e41958b5785826df2aecace0` - recommended version
+* `0xe314dfc40f0025322aacc0ba8ef420b62fb3b702cf01e0cdf3d829117ac2ff1a` - required version
+
+The correctness of these keys can be verified with `chisel --use 0.8.15  --evm-version london`.
+Just start it up and enter the slot definitions as found in the contract source code.
+```
+➜ bytes32(uint256(keccak256("protocolversion.recommended")) - 1)
+Type: bytes32
+└ Data: 0xe314dfc40f0025322aacc0ba8ef420b62fb3b702cf01e0cdf3d829117ac2ff1a
+➜ bytes32(uint256(keccak256("protocolversion.required")) - 1)
+Type: bytes32
+└ Data: 0x4aaefe95bd84fd3f32700cf3b7566bc944b73138e41958b5785826df2aecace0
+```
+
+Alternatively, `cast keccak` can be used.
+Call it with the storage slot string identifier, and subtract `1` form the result:
+```
+cast keccak protocolversion.recommended
+# 0xe314dfc40f0025322aacc0ba8ef420b62fb3b702cf01e0cdf3d829117ac2ff1b
+
+cast keccak protocolversion.required
+# 0x4aaefe95bd84fd3f32700cf3b7566bc944b73138e41958b5785826df2aecace1
+```
+
+## Version Encoding
+
+Each of the two aforementioned slots holds an _encoded_ protocol version that has the four typical
+semver components _major, minor, patch_ and _pre-release_.
+See [specs](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/superchain-upgrades.md#protocol-version) for more in-depth details.
+
+An encoded version for comparison can be genereated with the `protocol-version` encoding tool in the
+monorepo, e.g. for a bump of the major verion to 7:
+```
+cd $(git rev-parse --show-toplevel)/lib/optimism
+go run ./op-chain-ops/cmd/protocol-version encode --major 7
+```
+The tool can also be used to decode an encoded protocol version.
```

Comment:
Can we include a sample command here, or a link that shows how to do it?

Reply:
added a `decode` example.